### PR TITLE
feat(logs): add unit to metric filter

### PR DESCRIPTION
**Pull request by msambol**
Allows attaching unit to a metric filter.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by mcowgill-stripe** [comment]
the packages folder shouldn't be in there?

**Comment by msambol** [comment]
@TheRealAmazonKendra Can I ping you for a review here? :)

File: packages/@aws-cdk/aws-logs/lib/metric-filter.ts [hunk 1]
```diff
@@ -49,6 +49,7 @@ export class MetricFilter extends Resource {
         metricValue: props.metricValue ?? '1',
         defaultValue: props.defaultValue,
         dimensions: props.dimensions ? Object.entries(props.dimensions).map(([key, value]) => ({ key, value })) : undefined,
+        unit: props.unit ?? Unit.NONE,
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-logs/lib/metric-filter.ts:52]: If this were just the following, what would the CloudFormation behavior be?

```suggestion
        unit: props.unit,
```
I'm leaning toward not wanting the conditional of adding `Unit.NONE` to this but I want to make sure I understand what CloudFormation does in this case. Does it assign `Unit.NONE` under the hood?

In general, we don't want to make a change to a template if it isn't changing the behavior of that template. It concerns users when they see changes to their template without changing their code.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thank you for your contribution! I have one question/comment/suggestion inline.

**Comment by msambol** [comment]
@TheRealAmazonKendra Thanks for the feedback! I did confirm the unit is undefined when not specified (left below), so I removed the default of `NONE`. 

Screen Shot 2023-01-12 at 11 13 26 AM

**Review by TheRealAmazonKendra — APPROVED** [review]
This looks great! Thank you for the work on this!

**Pull request merged**